lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -928,6 +931,96 @@ static void checkOverrideAccessControl(ValueDecl *baseDecl, ValueDecl *decl,
   }
 }
 
+static GenericSignature *getNewGenericSignature(ValueDecl *base,
+                                                ValueDecl *derived) {
+  auto baseGenericCtx = base->getAsGenericContext();
+  auto derivedGenericCtx = derived->getAsGenericContext();
+  auto &ctx = base->getASTContext();
+
+  if (!baseGenericCtx) {
+    return nullptr;
+  }
+
+  auto baseClass =
+      baseGenericCtx->getInnermostTypeContext()->getSelfClassDecl();
+  auto derivedClass =
+      derivedGenericCtx->getInnermostTypeContext()->getSelfClassDecl();
+
+  if (!derivedClass) {
+    return nullptr;
+  }
+
+  if (!derivedClass->getInterfaceType()->is<LValueType>()) {
+    return nullptr;
+  }
+
+  auto subMap = derivedClass->getInterfaceType()->getContextSubstitutionMap(
+      derivedClass->getModuleContext(), baseClass);
+
+  auto *genericParams = baseGenericCtx->getGenericParams();
+  if (genericParams) {
+    SmallVector<GenericTypeParamDecl *, 4> newParams;
+
+    // First, clone the superclass constructor's generic parameter list,
+    // but change the depth of the generic parameters to be one greater
+    // than the depth of the subclass.
+    unsigned depth = 0;
+    if (auto *genericSig = baseClass->getGenericSignature())
+      depth = genericSig->getGenericParams().back()->getDepth() + 1;
+
+    for (auto *param : genericParams->getParams()) {
+      auto *newParam = new (ctx) GenericTypeParamDecl(
```

Comment:
You don't need to create new GenericTypeParamDecls. In fact it appears that newParams is not used below.